packages/foam-vscode/static/dataviz/graph.js

```diff
@@ -501,6 +502,15 @@ try {
       case 'didUpdateGraphData':
         graphData = augmentGraphInfo(message.payload);
         Actions.refreshWorkspaceData(graphData);
+        if (!graphUpdated) {
+          window.graphUpdated = true;
+          graph.zoom(graph.zoom() * 1.5);
+          graph.cooldownTicks(100);
+          graph.onEngineStop(() => {
+            graph.onEngineStop(() => {});
+            graph.zoomToFit(500);
+          });
+        }
```

Comment:
It's not easy to put this in `initDataviz` because `zoom()` depends on the number of nodes and, regarding the rest, one would have said that they should work that way but in practice it's not the case. Moreover, the API is not documented down to the required level of detail. This approach is safer and puts all the relevant code together in the same place.